graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsQueryExecutor.java

```diff
@@ -181,7 +181,23 @@ default DocumentContext executeAndGetDocumentContext(String query) {
      * @see <a href="https://github.com/json-path/JsonPath">JsonPath syntax docs</a>
      * @see <a href="https://graphql.org/learn/queries/#variables">Query Variables</a>
      */
-    <T> T executeAndExtractJsonPathAsObject(String query, String jsonPath, Map<String, Object> variables, Class<T> clazz);
+    default <T> T executeAndExtractJsonPathAsObject(String query, String jsonPath, Map<String, Object> variables, Class<T> clazz) {
+        return executeAndExtractJsonPathAsObject(query, jsonPath, null, Collections.emptyMap(), clazz);
+    }
+
+    /**
+     * Executes a GraphQL query, parses the returned data, extracts a value using JsonPath, and converts that value into the given type.
+     * Be aware that this method can't guarantee type safety.
+     * @param query Query string
+     * @param jsonPath JsonPath expression.
+     * @param headers Request headers represented as a Spring Framework {@link HttpHeaders}
+     * @param variables A Map of variables
+     * @param clazz The type to convert the extracted value to.
+     * @return The extracted value from the result, converted to type T
+     * @see <a href="https://github.com/json-path/JsonPath">JsonPath syntax docs</a>
+     * @see <a href="https://graphql.org/learn/queries/#variables">Query Variables</a>
+     */
+    <T> T executeAndExtractJsonPathAsObject(String query, String jsonPath, HttpHeaders headers, Map<String, Object> variables, Class<T> clazz);
```

Comment:
The position of the `header` argument is not consistent with other methods in the class. Could we add it as the last argument please.

* [here](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsQueryExecutor.java#L76)
* [and here](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsQueryExecutor.java#L158)

I consider this [here](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsQueryExecutor.java#L92) as an anomaly in the style.

Reply:
Thanks for the review. I addressed the comments and add some unit tests. Please take another look. Thanks.